src/lib.rs

```diff
@@ -565,6 +572,26 @@ pub fn throw(s: &str) -> ! {
     }
 }
 
+/// Rethrow a JS exception
+///
+/// This function will throw a JS exception with the JS value provided. This
+/// function will not return and the wasm stack will be popped until the point
+/// of entry of wasm itself.
+///
+/// Note that it is very easy to leak memory with this function because this
+/// function, unlike `panic!` on other platforms, **will not run destructors**.
+/// It's recommended to return a `Result` where possible to avoid the worry of
+/// leaks.
+#[cold]
+#[inline(never)]
+pub fn rethrow(s: JsValue) -> ! {
```

Comment:
nitpick: there is nothing about this function that requires that `s` was ever thrown before (I could be returning a result as the first throw) so this name is a bit off.

How about:

* `throw` -> `throw_str`
* `rethrow` -> `throw_val`

?

Reply:
Sounds good to me!